### PR TITLE
Add context awareness for jsonschema references across component descriptors

### LINKED ISSUE #303: Add context awareness for landscaper protocol schemes in JSON schema references
/area usability
/kind enhancement
/priority 5

**What would you like to be added**:

Imagine the following scenario:

**Component A**

blueprint
```
imports:
  name: foo
  type: data
  schema:
    $ref: "cd://resources/foo-definition"
```

JSON schema _foo-definition_
```
"properties": {
  "bar": {
    "$ref": "cd://resources/bar-definition"
  }
}
```

**Component B**

blueprint
```
imports:
  name: foo
  type: data
  schema:
    $ref: "cd://rcomponentReferences/component-a/resources/foo-definition"
```

This scenario would fail when instantiating blueprint of _component B_.
The reason is, that during the evaluation of the JSON schema of type _foo_, the context for reference `cd://resources/bar-definition` is no longer _component A_, but _component B_.
However, _component B_ has no own resource by the name _bar-definition_.

**Why is this needed**:

The usage JSON schemas of referenced components is a valid use case.
This means that Landscaper reference types should be supported.

### PR DESCRIPTION
**Pull request by Diaphteiros**
**How to categorize this PR?**

/area usability
/kind enhancement
/priority 3

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #303 

**Release note**:

```bugfix user
Resolving nested jsonschema references across components now works as intended.
```
```other developer
⚠️ Another dependency for testing has been added, please re-run `make install-requirements` before running `make test`.
```

### CODE REVIEW

**Review by achimweigel — APPROVED** [review]
only one small question about the complexity of a loop

**Comment by Diaphteiros** [comment]
Requires a release of the component-cli containing https://github.com/gardener/component-cli/pull/57

File: pkg/landscaper/jsonschema/reference.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+			sub, err := rr.resolve(v, subPath)
+			if err != nil {
+				return nil, err
+			}
+			resMap[k] = sub
+		}
+		if isRef {
+			return res, nil
+		}
+		return resMap, nil
```
[the remark this answers is not in the document]
Comment (Diaphteiros) [reply pkg/landscaper/jsonschema/reference.go]: I refactored this function and broke it down into smaller helper functions - please take a look again.

**Review by reshnm — APPROVED** [review]

**Pull request merged**